### PR TITLE
Use metatable information to support autocomplete; use NULL as the metadata field for valid columns with no extra associated information

### LINKED ISSUE #51: Use metatable information to support autocomplete
As @beckyjackson works on forms for editing rows with HTML forms (related to #35), it would be good to add support for autocomplete text fields. We tend to use Typeahead, which expects data in a simple JSON format, e.g. [Gizmos `search`](https://github.com/ontodev/gizmos#gizmossearch). The format is simple enough once we know what the available values are, and that should be guided by the metatable information. For a given field that we're editing:

1. if the datatype condition is `in("A", "B", "C")` then the autocomplete should provide: "A", "B", "C"
2. if the column structure is `from(foo.bar)`, the autocomplete should provide values from the `foo.bar` column
3. if the column structure is `under(foo.bar, "baz")`, the autocomplete should provide values from the "baz" branch of the `foo.bar` tree

So I think we need a function in the `validate` module that takes a table, column, and the string to complete, and returns a JSON array formatted for Typeagead.

### LINKED ISSUE #52: Consider using NULL to represent "valid" in _meta columns
When a cell is valid, the _meta column currently contains `json({"valid": true, "messages": []})`. To save space, and maybe reduce noise, should we replace these "valid" JSON objects with a simple NULL?

### PR DESCRIPTION
**Pull request by lmcmicu**
Closes #51, closes #52 (please close them manually since this PR is not going into master).

### CODE REVIEW

**Comment by lmcmicu** [comment]
@beckyjackson please look at `get_matching_values(config, table_name, column_name, leading_string="")` in `validate.py` and let me know if the function signature will work for you or if it needs to be tweaked.

File: src/script/validate.py [hunk 1]
```diff
@@ -33,6 +37,44 @@ def validate_row(config, table_name, row, existing_row=True, row_number=None):
...
+                raise ValidationException(f"No tree: '{table_name}.{tree_col}' found")
+            tree = tree[0]
+            under_val = structure["args"][1]["value"]
+            sql = with_tree_sql(tree, table_name, under_val) + "SELECT * FROM `tree`"
+            rows = config["db"].execute(sql)
+            values = [r[0] for r in rows.fetchall()]
+
+    values = [v for v in values if v.startswith(leading_string)]
+    values = [{"id": v, "label": v, "order": i} for i, v in enumerate(values, start=1)]
+    return json.dumps(values)
```
Comment (beckyjackson) [inline src/script/validate.py]: This should just `return values`, not the string

**Comment by beckyjackson** [comment]
Maybe I'm missing something, but I'm not able to get results from `get_matching_values` (just returns an empty list). I had to replace the code on lines 399-436 with what you sent me in Slack to get around multiprocessing, so maybe I'm missing a part of the loading.

Here's what I'm trying just to test getting values from the `table` column of the `table` table:

```
data = get_matching_values(config, "table", "table")
```

File: src/script/validate.py (hunk 1 above)
Comment (lmcmicu) [reply src/script/validate.py]: Will do.

**Comment by lmcmicu** [comment]
Hi Becky, for the example you mentioned:

`data = get_matching_values(config, "table", "table")`

this will not return anything because the datatype is not defined in terms of an `in()` condition. See the issue description here: https://github.com/jamesaoverton/cmi-pb-terminology/issues/51#issue-1140116403

We are only returning values for:

1. Datatypes defined in terms of "in"
2. "from" structures
3. "under" structures

As for `table_name`, it is defined in the `datatype` table as `exclude(/\W/)`

**Comment by lmcmicu** [comment]
To turn off the multiprocessing, set the `MULTIPROCESSING` flag to False (at [this line](https://github.com/jamesaoverton/cmi-pb-terminology/blob/3d8b8c12a6a21b6da1f2a1d14d34eac0e4be64ef/src/script/load.py#L39)). Sorry I should have set it to False by default. I'll do that now.

File: src/script/validate.py (hunk 1 above)
Comment (lmcmicu) [reply src/script/validate.py]: Done.

**Comment by beckyjackson** [comment]
Thanks - I got it working with Typeahead so the output looks good to me.

**Comment by lmcmicu** [comment]
Converting to draft since we need to modify the get_matching_values() function. Instead of only matching on values that begin with the given string parameter, we need to match on values that contain that parameter as a substring (which can be anywhere in the value). We should also perform the match in SQL instead of python. See [this issue comment](https://github.com/jamesaoverton/cmi-pb-terminology/issues/51#issuecomment-1048203655) for more info.

**Comment by jamesaoverton** [comment]
@beckyjackson We are no longer wrapping SQL JSON value with `json(...)` (which was a mistake I didn't understand/notice), so lines like this https://github.com/ontodev/sprocket/blob/standalone-table/sprocket/render.py#L262 `metadata = json.loads(meta[5:-1])` should be changed to `metadata = json.loads(meta)`

**Pull request merged**